Assets/Scripts/OrchestraArmy/Entity/Entities/Behaviour/AttackBehaviour.cs

```diff
@@ -99,6 +107,28 @@ public void Process(BehaviourStateMachine machine)
             });
             
         }
+        
+        /// <summary>
+        ///  calculates the vector of the note with player movement
+        /// </summary>
+        /// <param name="depth">acuracy of the aiming</param>
+        /// <param name="player">player entity to shoot</param>
+        /// <param name="enemyPosition">location of the enemy</param>
+        /// <param name="bulletSpeed"></param>
+        /// <param name="dirGuess">guess location interception</param>
+        /// <param name="timeGuess">guess time interception</param>
+        /// <returns></returns>
+        private Vector3 AimBot(int depth, Player player, Vector3 enemyPosition, float bulletSpeed, Vector3 dirGuess,float timeGuess = -1.0f)
+        {
+            if (timeGuess == -1.0f)
```

Comment:
floating-point comparison is niet handig op deze manier, vaak krijg je dan false-negatives

Reply:
Mathf.Approximately ofzo is idd beter, maar omdat hij hier altijd met constanten werkt (default van -1.0f in beide gevallen) zou het wel goed moeten gaan

Wel risky though, dus beter om aan te passen just to be sure